django/forms/models.py

```diff
@@ -1112,7 +1112,7 @@ def __iter__(self):
             yield self.choice(obj)
 
     def __len__(self):
-        return (len(self.queryset) + (1 if self.field.empty_label is not None else 0))
+        return (len(self.queryset.all()) + (1 if self.field.empty_label is not None else 0))
```

Comment:
An optimized `__bool__` definitely makes sense.

On the other hand (as mentioned in the ticket), the queryset is (now, since 1.11) only shared per form instance, and not per form class. Given that, an alternative approach would be to remove the `.all()` in `__iter__`, and indeed reuse the queryset for the lifetime of the form-instance.